gtk/src/toga_gtk/widgets/textinput.py

```diff
@@ -24,6 +25,11 @@ def gtk_focus_out_event(self, entry, user_data):
         if self.interface.on_lose_focus:
             self.interface.on_lose_focus(self.interface)
 
+    def gtk_key_release_event(self, key, user_data):
+        if user_data.keyval == 65293 or user_data.keyval == 65421:
+            if self.interface.on_return:
```

Comment:
There's no need to perform this check - events *always* have a handler, even if it's a no-op.